doc/modules/ensemble.rst

```diff
@@ -838,7 +817,163 @@ accessed via the ``feature_importances_`` property::
+the estimator on a validation set.
+
+Missing values support
+----------------------
+
+:class:`HistGradientBoostingClassifier` and
+:class:`HistGradientBoostingRegressor` have built-in support for missing
+values (NaNs).
+
+During training, the tree grower learns at each split point whether samples
+with missing values should go to the left or right child, based on the
+potential gain. When predicting, samples with missing values are assigned to
+the left or right child consequently.
+
+
+.. TODO: Add this example when missing values PR is merged (results are
+.. wrong for now)
+
+.. from sklearn.experimental import enable_hist_gradient_boosting
+.. from sklearn.ensemble import HistGradientBoostingRegressor
+.. import numpy as np
+
+.. X = np.array([0, 1, 2, np.nan]).reshape(-1, 1)
+.. y = [0, 0, 1, 1]
+.. gbdt = HistGradientBoostingRegressor().fit(X, y)
+.. gbdt.predict(X)
+
+If no missing values were encountered for a given feature during training,
+then samples with missing values are mapped to whichever child has the most
+samples.
+
+Low-level parallelism
+---------------------
+
+:class:`HistGradientBoostingClassifier` and
+:class:`HistGradientBoostingRegressor` have implementations that use OpenMP
+for parallelization through Cython. The number of threads that is used can
+be changed using the ``OMP_NUM_THREADS`` environment variable. By default,
+all available cores are used. Please refer to the OpenMP documentation for
+details.
+
+Why it's faster
+---------------
+
+The bottleneck of a gradient boosting procedure is building the decision
+trees. Building a traditional decision tree (as in the other GBDTs
+:class:`GradientBoostingClassifier` and :class:`GradientBoostingRegressor`)
+requires sorting the samples at each node (for
+each feature). Sorting is needed so that the potential gain of a split point
+can be computed efficiently. Splitting a single node has thus a complexity
+of :math:`\mathcal{O}(\text{n_features} * n \log(n))` where :math:`n` is the
+number of samples at the node.
+
+:class:`HistGradientBoostingClassifier` and
+:class:`HistGradientBoostingRegressor`, in contrast, do not require sorting the
+feature values and instead use a data-structure called a histogram, where the
+samples are implicitly ordered. Building a histogram has a
+:math:`\mathcal{O}(n)` complexity, so the node splitting procedure has a
+:math:`\mathcal{O}(\text{n_features} * n)` complexity, much smaller than the
+previous one. In addition, instead of considering :math:`n` split points, we
+here consider only ``max_bins`` split points, which is much smaller.
+
+In order to build histograms, the input data `X` needs to be binned into
+integer-valued bins. This binning procedure does require sorting the feature
+values, but it only happens once at the very beginning of the boosting process
+(not at each node, like in :class:`GradientBoostingClassifier` and
+:class:`GradientBoostingRegressor`).
+
+Finally, many parts of the implementation of
```

Comment:
There are four places, we can list them?

1. Binning
2. Building Histograms
3. Splitting
3. Predicting

Reply:
good idea